backend/src/main/java/com/aadeshandreas/ailearning/ai_learning_companion/controller/DocumentController.java

```diff
@@ -54,6 +55,30 @@ public ResponseEntity<ApiResponse<?>> uploadDocument(@RequestParam("file") Multi
         }
     }
 
+    public ResponseEntity<ApiResponse<?>> uploadText(@RequestBody TextUploadRequest request){
+        try {
+            if(request.text() == null || request.text().trim().isEmpty()){
+                ApiResponse<Void> errorResponse = new ApiResponse<>(
+                        "Text cannot be empty",
+                        "INVALID_INPUT",
+                        null
+                );
+                return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(errorResponse);
+            }
+            documentService.uploadText(request.text());
+            return ResponseEntity.ok(new ApiResponse<>("Success", "200_OK", null));
+        }catch (Exception e){
+            logger.error("Error processing text upload" , e);
+
+            ApiResponse<Void> errorResponse = new ApiResponse<>(
+                    "Unable to process text",
+                    "TEXT_UPLOAD_ERROR",
+                    null
+            );
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(errorResponse);
+        }
+    }
```

Comment:
```suggestion
    @PostMapping(value = "/upload-text")
    public ResponseEntity<ApiResponse<?>> uploadText(@Valid @RequestBody DocumentText documentText){
        try {
            documentService.uploadText(documentText.text());
            return ResponseEntity.ok(new ApiResponse<>("Success", "200_OK", null));
        } catch (Exception e){
            logger.error("Error processing text upload", e);

            ApiResponse<Void> errorResponse = new ApiResponse<>(
                    "Unable to process text",
                    "TEXT_UPLOAD_ERROR",
                    null
            );
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(errorResponse);
        }
    }
```

You forgot to add the @PostMapping which ensures that spring boot can present the correct api to the client.
Your DTO already does validation check so we can just wrap that logic inside of a try catch in the beginning that way the controller only handles the service calls and exceptions.
I also changed the name of the DTO class to a noun for consistency.

We can use @Valid to make it work with my comment on your DTO